src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -11,25 +11,48 @@
   (not= (get-in req1 [:params "query"])
         (get-in req2 [:params "query"])))
 
-(defn add-criteria [crit query]
+(defn query-criteria
+  "Extract the 'criteria' (select) part of the given query"
+  [query]
   (cm/match [query]
-            [["extract" columns (expr :guard nil?)]]
-            ["extract" columns crit]
+    [["extract" _ expr]] expr
+    :else (or query nil)))
+
+(defn is-active-node-criteria? [criteria]
+  (cm/match [criteria]
+    [["=" ["node" "active"] _]] criteria
+    :else false))
```

Comment:
I'm not sure if we should or not. My impression from the query engine was
that `[= [node active] true]` was special, so I wouldn't have to worry
about the general case.

On Fri, May 8, 2015 at 1:25 PM, Wyatt Alt <notifications@github.com> wrote:

> In src/puppetlabs/puppetdb/http/query.clj
> <https://github.com/puppetlabs/puppetdb/pull/1377#discussion_r29973518>:
>
> >    (cm/match [query]
> > -            [["extract" columns (expr :guard nil?)]]
> > -            ["extract" columns crit]
> > +    [["extract" _ expr]] expr
> > +    :else (or query nil)))
> > +
> > +(defn is-active-node-criteria? [criteria]
> > +  (cm/match [criteria]
> > +    [["=" ["node" "active"] _]] criteria
> > +    :else false))
>
> I should say, we say we support them. They currently don't work on the
> deactivated field due to restrictions.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppetdb/pull/1377/files#r29973518>.
>